### PR TITLE
[CUDA] BF16 support

### PR DESCRIPTION
**Pull request by hgt312**
https://discuss.tvm.apache.org/t/cuda-bfloat16-support/8567
@junrushao1994 @merrymercy @yzhliu

### CODE REVIEW

**Comment by tqchen** [comment]
@yzhliu @hgt312  can we please followup on this PR?

cc @vinx13 @masahi

File: tests/python/unittest/test_target_codegen_cuda.py [hunk 1]
```diff
@@ -905,26 +952,27 @@ def test_unrolled_vectorization():
...
+    # test_cuda_mix_threaded_and_normal_reduction()
+    # test_cuda_floordiv_with_vectorization()
+    # test_cuda_floormod_with_vectorization()
+    # test_vectorized_intrin1()
+    # test_vectorized_intrin2()
+    # test_vectorized_popcount()
+    # test_cuda_vectorize_load_permute_pad()
+    # test_vectorized_cooperative_fetching_x()
+    # test_vectorized_cooperative_fetching_xy()
+    # test_unrolled_vectorization()
```
Comment (jcf94) [inline tests/python/unittest/test_target_codegen_cuda.py:978]: Uncommon these tests.

**Review by jcf94 — CHANGES_REQUESTED** [review]
Plz rebase this branch

**Comment by jcf94** [comment]
> @yzhliu @hgt312 can we please followup on this PR?
> 
> @jcf94 would be great if you can help to manage this PR (also cc @vinx13 @masahi)

@tqchen Thanks! I'd love to help.

If @hgt312 don't have time to finish these remaining issues, I can help clean them this weekend.

**Review by jcf94 — APPROVED** [review]
Thanks! @hgt312

**Pull request merged**